src/main.c

```diff
@@ -922,11 +942,19 @@ static int open_unix_listener(h2o_configurator_command_t *cmd, yoml_t *node, str
     }
 
     /* add new listener */
-    if ((fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1 || bind(fd, (void *)sa, sizeof(*sa)) != 0 || listen(fd, H2O_SOMAXCONN) != 0) {
+    if (
+#if defined(__linux__)
+        (fd = socket(AF_UNIX, SOCK_STREAM | SOCK_NONBLOCK | SOCK_CLOEXEC, 0)) == -1
+#else
+        (fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1
+#endif
+        || bind(fd, (void *)sa, sizeof(*sa)) != 0 || listen(fd, H2O_SOMAXCONN) != 0) {
```

Comment:
Do we need to change this? I think it's irrevelant to the intent of the PR.

Reply:
will be reverted。